tests/test_grainstats.py

```diff
@@ -151,3 +152,25 @@ def test_random_grain_stats(caplog, tmpdir) -> None:
     grainstats.calculate_stats()
 
     assert "No labelled regions for this image, grain statistics can not be calculated." in caplog.text
+
+
+@pytest.mark.parametrize("coords, shape, expected", [
+    (np.asarray([5,5]),10,0),
+    (np.asarray([-3,12]),10,3),
+    (np.asarray([-3,14]),10,-4)
+    ])
+def test_get_shift(coords, shape, expected):
+    assert GrainStats.get_shift(coords, shape) == expected
```

Comment:
Linting will complain about a lack of docstring here. I often feel its a little futile and so keep it simple and say something like.

```
    """Test the GrainStats.get_shift() method."""
```